File: leetcode/easy/3079-find-the-sum-of-encrypted-integers/3079-find-the-sum-of-encrypted-integers.py
# Definition for singly-linked list.
from typing import List


class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        strNums = list(map(str, nums))
        result = []

        for strNum in strNums:
            length = len(strNum)
            maxNum = max(strNum)
            result.append(int(maxNum * length))

        return sum(result)

File: leetcode/easy/3079-find-the-sum-of-encrypted-integers/3079-find-the-sum-of-encrypted-integers.kt
class Solution {
    fun sumOfEncryptedInt(nums: IntArray): Int {
        val strNums = nums.map { it.toString().chunked(1) }
        val maxStrs = strNums.map {
            it.max()
                .repeat(it.size)
                .toInt()
        }

        return maxStrs.sum()
    }
}
